ethers-etherscan/src/contract.rs

```diff
@@ -159,6 +173,19 @@ impl ContractMetadata {
     }
 }
 
+fn libs_serialize<S>(l: &Option<Libraries>, serializer: S) -> std::result::Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let libraries = &l.as_ref().unwrap().libs;
+    let mut map = serializer.serialize_map(Some(libraries.len()))?;
+    for (i, (k, v)) in Vec::from_iter(libraries.iter()).iter().enumerate() {
+        map.serialize_entry(&format!("libraryname{}", i), &k.split(":").last())?;
+        map.serialize_entry(&format!("libraryaddress{}", i), v)?;
+    }
```

Comment:
this should only serialize `if let Some(ref libs) =  l = {...}`
could you an example of what this should serialize to?

Reply:
thanks, appreciate the feedback! still trying to understand control flow in rust. i'll update this to reflect your suggestion. i will also add a basic serialization test as an example. just need to find some time 😅